### ISSUE TITLE
CSL is not found when defined in an extension. 

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#3944]
I'm having a related bug when the csl is defined in an extension. Both in 1.2.313 and 1.3.97 on Windows. It works fine on Ubuntu.

```
File _extensions-institute-for-nature-and-forest.csl not found in resource path
ERROR: File _extensions-institute-for-nature-and-forest.csl not found in resource path
```

The report and extension are available at https://github.com/inbo/inboqmd_examples/tree/main/source/report/en

_Originally posted by @ThierryO in https://github.com/quarto-dev/quarto-cli/issues/2945#issuecomment-1380175605_

**Comment by cderv** [comment]
@ThierryO I can reproduce with your source code (by commenting out fonts because Inconsolata not found on my side but anyway). 

I'll have a closer look to this.

**Comment by cderv** [comment]
So this is directly link to previous issue #2945 with same cause : 

`csl` in YAML will be parsed as Markdown. This means this applies https://github.com/quarto-dev/quarto-cli/issues/2945#issuecomment-1289011178 : `\` in `_extensions\report\research-institute-for-nature-and-forest.csl` will mis interpreted which lead to the issue. 

Dummy example 

````markdown
---
title: "test-quarto-project"
bibliography: common.bib
csl: _extensions\report\research-institute-for-nature-and-forest.csl
---

Content
````

You see below how the Meta is parsed 
````
> quarto pandoc --to native -s .\test-quarto-project.qmd
Pandoc                                                                   
	Meta                                                                     
		{ unMeta =roject.qmd      
	        fromList
	          [ ( "bibliography" , MetaInlines [ Str "common.bib" ] )      
	          , ( "csl"
	            , MetaInlines
	                [ Str "_extensions"
	                , RawInline (Format "tex") "\\report"
	                , RawInline (Format "tex") "\\research"
	                , Str "-institute-for-nature-and-forest.csl"
	                ]
	            )
	          , ( "title" , MetaInlines [ Str "test-quarto-project" ] )    
	          ]
	    }
  [ Para [ Str "Content" ] ]
````

This is Pandoc will then throw the error with a wrong path.
````
File _extensions-institute-for-nature-and-forest.csl not found in resource path
````

Thanks for the report @ThierryO 

cc @dragonstyle

**Comment by cderv** [comment]
Trying to understand the previous fix though

We did d78d4b22966df1a122d8e0bd13219ec7956a5005 it seems as absolute path was needed. 
But then we did a91bd59aa9a6eafe3e6d47873019bb31b4c3edac that makes it again relative. 

We get the wrong slash in the process of making absolute.

**Comment by cderv** [comment]
@cscheid about https://github.com/quarto-dev/quarto-cli/issues/2945#issuecomment-1380480019

Yes the example as backslashed to show why we get the error. It is meant to explain why this is not working currently. 
Our code introduce this because of `join()` on Windows in Deno. 
* Extension as forward slash for `csl`
* When quarto handle bibliography, `csl` is retrieved and made absolute -> `/` became `\`
* We put back this path in relative `csl` in yaml header
* pandoc fails because it is wrongly parsed.

I'll push a PR so that you can see my suggested fix. I can explain in detail live if we want to revisit this relative -> absolute -> relative conversion mentioned above.

**Comment by cderv** [comment]
@ThierryO This should now be fixed in v1.3.121

**Comment by ThierryO** [comment]
@cderv tested with v1.3.124 on Ubuntu and Windows. I confirm that the issue is fixed.
